pkg/util/runtime/runtime_linux.go

```diff
@@ -29,7 +31,20 @@ func parseKernelVersionStr(kernelVersionStr string) (semver.Version, error) {
 	// Patch: 0
 	// Pre: [72-generic]
 	// Build: []
-	return semver.Parse(kernelVersionStr)
+	verStrs := strings.Split(kernelVersionStr, ".")
+	switch {
+	case len(verStrs) < 2:
+		return semver.Version{}, fmt.Errorf("unable to get kernel version from %q", kernelVersionStr)
+	case len(verStrs) < 3:
+		verStrs = append(verStrs, "0")
+	}
+	patch := regexp.MustCompilePOSIX(`^[0-9]+`).FindString(verStrs[2])
+	if patch == "" {
+		verStrs[2] = "0"
+	} else {
+		verStrs[2] = patch
+	}
+	return semver.Parse(strings.Join(verStrs[:3], "."))
```

Comment:
I feel the above has a little redundant logic as semver.Parse. If we are sure that a kernel version is at least in this format "x.y.z-????", maybe it can just `semver.Parse(strings.SplitN(kernelVersionStr, "-", 2)[0])`.
But the comment should be adjusted as well.

Reply:
It will also lose prerelease value, if we use `strings.SplitN(kernelVersionStr, "-", 2)[0]`. and maybe there does not has prerelease in kernel version string, like: "4.19.43.el7.x86_64".  `strings.SplitN(kernelVersionStr, "-", 2)[0]` will return full string.

Reply:
> I feel the above has a little redundant logic as semver.Parse. If we are sure that a kernel version is at least in this format "x.y.z-????", maybe it can just `semver.Parse(strings.SplitN(kernelVersionStr, "-", 2)[0])`.
> But the comment should be adjusted as well.

It will also lose prerelease value, if we use` strings.SplitN(kernelVersionStr, "-", 2)[0]`. and maybe there does not has prerelease in kernel version string, like: "4.19.43.el7.x86_64". `strings.SplitN(kernelVersionStr, "-", 2)[0]` will return full string.

Reply:
I see, thanks for the explantion.